controller/main.go

```diff
@@ -88,7 +88,9 @@ func (c *controller) SetBalancer(l log.Logger, name string, svcRo *v1.Service, _
 
 	if !reflect.DeepEqual(svcRo.Status, svc.Status) {
 		var st v1.ServiceStatus
-		st, svc = svc.Status, svcRo.DeepCopy()
+		var annotations map[string]string
+		st, annotations, svc = svc.Status, svc.Annotations, svcRo.DeepCopy()
+		svc.Annotations = annotations
```

Comment:
I think we should reflect.DeepCopy annotations here, as what you are doing is copying the reference of the map (which is owned by svcRo)

Reply:
Also, given this is becoming a bit beefy, consider moving it in a function

Reply:
Thanks, Done.

Reply:
Please review the latest changes, I appreciate it if you give me some advices